extras/static_code_analysis.sh

```diff
@@ -165,9 +168,17 @@ for f in $FILE_NAMES; do
         tail -n +2 "${f_base}_cppcheck.txt" > "${f_base}_cppcheck.tmp" && mv "${f_base}_cppcheck.tmp" "${f_base}_cppcheck.txt"
         if [ -s "${f_base}_cppcheck.txt" ]; then
           cppcheck_failed=true
+          typeset -i msg_count=0
           cat ${f_base}_cppcheck.txt | while read line
           do
             print_msg "MSGBLD0155: " "[CPPC] $line"
+            if $RUNS_ON_TRAVIS; then
+              msg_count+=1
+              if [ ${msg_count} -ge ${MAX_CPPCHECK_MSG_COUNT} ]; then 
+                print_msg "MSGBLD0156: " "[CPPC] MAX_CPPCHECK_MSG_COUNT (${MAX_CPPCHECK_MSG_COUNT}) reached for file: $f"
+                break
+              fi
+            fi
           done
```

Comment:
I suggest the following code (without explicit counter) as a replacement for your lines 171-181:
```bash
head -n ${MAX_CPPCHECK_MSG_COUNT} ${f_base}_cppcheck.txt | while read line; do
  print_msg "MSGBLD0155: " "[CPPC] $line"
done

if ${RUNS_ON_TRAVIS} && [ `wc -l ${f_base}_cppcheck.txt` -ge ${MAX_CPPCHECK_MSG_COUNT} ]; then
  print_msg "MSGBLD0156: " "[CPPC] MAX_CPPCHECK_MSG_COUNT (${MAX_CPPCHECK_MSG_COUNT}) reached for file: $f"
fi
```